core/src/sqlite_workers/sqlite_database.rs

```diff
@@ -1,249 +1,228 @@
-        let runtime_handle = tokio::runtime::Handle::current();
-
-        task::spawn_blocking(move || {
-            let conn =
-                create_in_memory_sqlite_db(runtime_handle, databases_store, database_id, &tables)
-                    .unwrap();
-
-            while let Some(message) = rx.blocking_recv() {
-                match message {
-                    DbMessage::Execute { query, response } => {
-                        let _ = response.send(Ok(execute_query_on_conn(&conn, query)?));
-                    }
-                }
+    ) -> Result<()> {
+        match &self.conn {
+            Some(_) => Ok(()),
+            None => {
+                self.conn = Some(
+                    create_in_memory_sqlite_db(databases_store, &self.database_id, &tables).await?,
+                );
+
+                Ok(())
             }
-
-            Ok::<(), anyhow::Error>(())
-        });
-
-        Self { sender: tx }
+        }
     }
 
     pub async fn query(&self, query: String) -> Result<Vec<DatabaseResult>> {
-        // Create a oneshot channel for the response
-        let (response_tx, response_rx) = oneshot::channel();
-
-        // Send the query and the sender part of the oneshot channel
-        self.sender
-            .send(DbMessage::Execute {
-                query,
-                response: response_tx,
-            })
-            .await
-            .unwrap();
-
-        // Await the response
-        match response_rx.await {
-            Ok(result) => Ok(result?),
-            Err(e) => Err(anyhow!("Failed to receive response: {}", e)),
+        match &self.conn {
+            Some(conn) => execute_query_on_conn(conn.clone(), query).await,
+            None => Err(anyhow!("Database not initialized")),
         }
     }
 }
 
-fn create_in_memory_sqlite_db(
-    runtime_handle: Handle,
+async fn create_in_memory_sqlite_db(
     databases_store: Box<dyn DatabasesStore + Sync + Send>,
-    database_id: String,
+    database_id: &str,
     tables: &Vec<DatabaseTable>,
-) -> Result<Connection> {
-    async fn fetch_rows(
-        database_id: &str,
-        table: &DatabaseTable,
-        databases_store: Box<dyn DatabasesStore + Sync + Send>,
-    ) -> Result<(DatabaseTable, Vec<DatabaseRow>)> {
-        let (rows, _) = databases_store
-            .list_database_rows(&database_id, table.table_id(), None)
-            .await?;
-
-        Ok((table.clone(), rows))
-    }
-
+) -> Result<Arc<Mutex<Connection>>> {
     let time_get_rows_start = utils::now();
 
-    let rows = runtime_handle.block_on(async move {
-        try_join_all(
-            tables
-                .iter()
-                .map(|table| fetch_rows(&database_id, table, databases_store.clone())),
-        )
-        .await
-    })?;
+    let rows = try_join_all(tables.iter().map(|table| {
+        let table_clone = table.clone();
```

Comment:
Couldn't find a way to not clone that. Shouldn't be too bad but isn't super satisfying.